src/js/plugins/plugin.text_selection.js

```diff
@@ -0,0 +1,204 @@
+//@ts-check
+import { isFirefox } from './tts/utils';
+
+const BookReader = /** @type {typeof import('../BookReader').default} */(window.BookReader);
+
+class TextSelectionPlugin {
+
+  constructor(avoidTspans = isFirefox()) {
+    /**@type {PromiseLike<JQuery<HTMLElement>>} */
+    this.djvuPagesPromise = null;
+    // Using text elements insted of tspans for words because Firefox does not allow svg tspan strech.
+    // Tspans are necessary on Chrome because they prevent newline character after every word when coping
+    this.svgParagraphElement = "text";
+    this.svgWordElement = "tspan";
+    this.onFirefox = avoidTspans
+    if(avoidTspans) {
+      this.svgParagraphElement = "g";
+      this.svgWordElement = "text";
+    }
+  }
+
+  /**
+   * @param {string} ocaid
+   */
+  init(ocaid) {
+    this.djvuPagesPromise = $.ajax({
+      type: "GET",
+      url: `https://cors.archive.org/cors/${ocaid}/${ocaid}_djvu.xml`,
+      dataType: "xml",
+
+      error: function (e) {
+        return undefined;
+      }
+    }).then(function (xmlMap) {
+      if (xmlMap != undefined) {
+        return $(xmlMap).find("OBJECT");
+      }
+    });
+  }
+
+  /**
+   * @param {number} index
+   * @returns {Promise<HTMLElement>}
+   */
+  async getPageText(index) {
+    return (await this.djvuPagesPromise)[index];
+  }
+
+  /**
+   * Intercept copied text
+   * @param {JQuery} $container
+   */
+  interceptCopy($container) {
```

Comment:
Can we attach this to the container directly instead of to each page?

Reply:
`.clipboardData` does not seem to work with jquery events, only with HTML ones

Reply:
So `$container[0].addEventListener( ...`; or you can get the original, HTML event from the jquery event using `event.originalEvent` https://stackoverflow.com/questions/5926123/get-the-raw-event-from-jquery

Reply:
done